src/server/routes/oauth.ts

```diff
@@ -0,0 +1,185 @@
+import { typedRouter as router } from '@/server/lib/typedRoutes';
+import { Request } from 'express';
+import { ResponseWithRequestState } from '@/server/models/Express';
+import {
+  deleteAuthorizationStateCookie,
+  getAuthorizationStateCookie,
+  OpenIdErrors,
+  ProfileOpenIdClient,
+  ProfileOpenIdClientRedirectUris,
+} from '@/server/lib/okta/openid-connect';
+import { getConfiguration } from '@/server/lib/getConfiguration';
+import { logger } from '@/server/lib/serverSideLogger';
+import { trackMetric } from '@/server/lib/trackMetric';
+import { handleAsyncErrors } from '@/server/lib/expressWrappers';
+import { exchangeAccessTokenForCookies } from '@/server/lib/idapi/auth';
+import { setIDAPICookies } from '@/server/lib/setIDAPICookies';
+import { SignInErrors } from '@/shared/model/Errors';
+import {
+  readEncryptedStateCookie,
+  setEncryptedStateCookie,
+} from '@/server/lib/encryptedStateCookie';
+import { addQueryParamsToPath } from '@/shared/lib/queryParams';
+
+interface OAuthError {
+  error: string;
+  error_description: string;
+}
+
+const { defaultReturnUri } = getConfiguration();
+
+/**
+ * Type guard to check that a given error is an OAuth error.
+ * By checking for the `error` and `error_description` properties
+ * @param {unknown} obj
+ * @return {boolean}
+ */
+const isOAuthError = (
+  maybeOAuthError: unknown,
+): maybeOAuthError is OAuthError => {
+  const { error, error_description } = maybeOAuthError as OAuthError;
+  return error !== undefined && error_description !== undefined;
+};
+
+/**
+ * Helper method to redirect back to the sign in page with
+ * a generic error message if we don't want to expose the error
+ * back to the client. Be sure to log the error though!
+ */
+const redirectForGenericError = (
+  req: Request,
+  res: ResponseWithRequestState,
+) => {
+  // we set the sign in redirect to true so that we can render the sign in page
+  // without getting into a redirect loop
+  const encryptedState = readEncryptedStateCookie(req);
+  setEncryptedStateCookie(res, {
```

Comment:
Is it worth updating these two calls to be `updateEcryptedStateCookie(req: Request, res: ResponseWithRequestState ,state: EncryptedState)` and you can perform the  `readEncryptedStateCookie` inside the update function ?